### ISSUE TITLE
Implement out of context unwinding for OSX

### ISSUE THREAD

**Issue by mikem8361** [dotnet/coreclr#1787]
The DAC needs out of context/process stack unwinding for the special method helper frame.  This is implemented for Linux distributions that has the libunwind-ptrace support and not for OSX.

For OSX (and probably for the other Linux distributions), we need to implement a mach-o format reader  to get the stack unwind info. This is will take from 5-8 days.

**Comment by gkhanna79** [comment]
@noahfalk @cshung

**Comment by jkotas** [comment]
cc @smosier you may be interested in this

**Comment by Matt-Esch** [comment]
I came across this issue while looking for a solution to out of context unwinding on OSX for producing flame graphs. I came to the conclusion I would have to implement a mach-o based address space object and pass it to libunwind.

However this work has already been done for lldb on OSX so you might be able to adapt this to your needs:
https://github.com/llvm-mirror/libunwind/blob/master/src/libunwind.cpp#L112

Where I'm using this to create the task:
```c
kern_return_t attach_process(pid_t pid, task_t* port) {
    return task_for_pid(mach_task_self(), pid, port);
}
```

**Comment by mikem8361** [comment]
Thanks. I’m currently looking into the llvm libunwind.

**Comment by tzwlai** [comment]
This has been addressed by #2260.

CC @mikem8361
